hail/python/hailtop/auth/tokens.py

```diff
@@ -17,8 +17,7 @@ def get_tokens_file():
             return os.path.expanduser('~/.hail/tokens.json')
         return '/user-tokens/tokens.json'
 
-    def __init__(self):
-        tokens_file = self.get_tokens_file()
+    def __init__(self, tokens_file):
         if os.path.isfile(tokens_file):
             with open(tokens_file, 'r') as f:
                 self._tokens = json.load(f)
```

Comment:
If you specify a tokens file and it doesn't exist, that feels like it should be an error.

Reply:
What do you think of this?
```python3
@staticmethod
def default_tokens():
    default_tokens_file = self.get_tokens_file()
    if os.path.isfile(default_tokens_file):
        return Tokens.tokens_from_file(default_tokens_file)
    return Tokens({})

@staticmethod
def tokens_from_file(fname):
    with open(fname, 'r') as f:
        return Tokens(json.load(f))

def __init__(self, tokens):
    self._tokens = tokens
```

Reply:
Yeah, that makes sense. I'll change it.